apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerOverviews.cs

```diff
@@ -0,0 +1,283 @@
+                                });
+                            cmd.Transaction = transaction;
+
+                            // This executes a single transactions with alls row written at once during CommitAsync().
+                            // If a transient fault occurs, this entire method is re-run.
+                            for (var i = 0; i < 5; i++)
+                            {
+                                cmd.Parameters["Key"].Value = Guid.NewGuid().ToString("N");
+                                cmd.Parameters["StringValue"].Value = $"StringValue{i}";
+                                cmd.Parameters["Int64Value"].Value = i;
+                                await cmd.ExecuteNonQueryAsync();
+                            }
+
+                            await transaction.CommitAsync();
+                        }
+                    }
+                });
+            // End sample
+        }
+
+#if NET451 || NET451 || NET452
+        [Fact]
+        public async Task TransactionScopeAsync()
+        {
+            await _fixture.EnsureTestDatabaseAsync().ConfigureAwait(false);
+
+            // Sample: TransactionAsync
+            var retryPolicy = new RetryPolicy<SpannerFaultDetectionStrategy>(RetryStrategy.DefaultExponential);
+
+            await retryPolicy.ExecuteAsync(
+                async () =>
+                {
+                    using (var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+                    {
+                        using (var connection =
+                            new SpannerConnection(
+                                $"Data Source=projects/{_projectid}/instances/{_instanceName}/databases/{_databaseName}"))
+                        {
+                            await connection.OpenAsync();
+
+                            var cmd = connection.CreateInsertCommand(
+                                "TestTable", new SpannerParameterCollection
+                                {
+                                    {"Key", SpannerDbType.String},
+                                    {"StringValue", SpannerDbType.String},
+                                    {"Int64Value", SpannerDbType.Int64}
+                                });
+
+                            // This executes a single transactions with alls row written at once during scope.Complete().
+                            // If a transient fault occurs, this entire method is re-run.
+                            for (var i = 0; i < 5; i++)
+                            {
+                                cmd.Parameters["Key"].Value = Guid.NewGuid().ToString("N");
+                                cmd.Parameters["StringValue"].Value = $"StringValue{i}";
+                                cmd.Parameters["Int64Value"].Value = i;
+                                await cmd.ExecuteNonQueryAsync();
+                            }
+
+                            scope.Complete();
+                            await Task.Yield();
```

Comment:
I'm hoping to remove this soon (before we go GA).  Explaining it will be pretty tough.  I'll add a TODO